aws/setup.go

```diff
@@ -0,0 +1,307 @@
+package aws
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/e2etemplates/pkg/chartvalues"
+	"github.com/giantswarm/e2etemplates/pkg/e2etemplates"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/e2esetup/aws/env"
+)
+
+const (
+	awsOperatorArnKey     = "aws.awsoperator.arn"
+	awsResourceValuesFile = "/tmp/aws-operator-values.yaml"
+	credentialName        = "credential-default"
+	credentialNamespace   = "giantswarm"
+	provider              = "aws"
+)
+
+func Setup(ctx context.Context, m *testing.M, config Config) error {
+	var v int
+	var err error
+	var errors []error
+
+	if config.AWSClient == nil {
+		return microerror.Maskf(invalidConfigError, "%T.AWSClient must not be empty", config)
+	}
+	if config.Guest == nil {
+		return microerror.Maskf(invalidConfigError, "%T.Guest must not be empty", config)
+	}
+	if config.Host == nil {
+		return microerror.Maskf(invalidConfigError, "%T.Host must not be empty", config)
+	}
+
+	vpcPeerID, err := installHostPeerVPC(config)
+	if err != nil {
+		errors = append(errors, err)
+		v = 1
+	}
+
+	err = config.Host.Setup()
+	if err != nil {
+		errors = append(errors, err)
+		v = 1
+	}
+
+	err = installResources(config, vpcPeerID)
+	if err != nil {
+		errors = append(errors, err)
+		v = 1
+	}
+
+	if v == 0 {
+		err = config.Guest.Setup()
+		if err != nil {
+			errors = append(errors, err)
+			v = 1
+		}
+	}
+
+	if v == 0 {
+		v = m.Run()
+	}
+
+	if env.KeepResources() != "true" {
+		config.Host.DeleteGuestCluster(ctx, provider)
+
+		// only do full teardown when not on CI
+		if env.CircleCI() != "true" {
+			err := teardown(config)
+			if err != nil {
+				errors = append(errors, err)
+				v = 1
+			}
+			// TODO there should be error handling for the framework teardown.
+			config.Host.Teardown()
+		}
+	}
+
+	if len(errors) > 0 {
+		return microerror.Mask(errors[0])
+	}
+
+	return nil
+}
+
```

Comment:
Rest of the code comes from aws-operator. There are more changes I'd like to make but I'll do it once the library is used by aws-operator and cluster-operator.